pkg/app/init.go

```diff
@@ -0,0 +1,135 @@
+package app
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"go.uber.org/zap"
+	"helm.sh/helm/v3/pkg/cli"
+
+	"github.com/helmfile/helmfile/pkg/helmexec"
+)
+
+const (
+	HelmInstallCommand      = "curl -fsSL https://raw.githubusercontent.com/helm/helm/main/scripts/get-helm-3 | bash"
```

Comment:
1. It depends curl command. maybe curl not be installed.
2. We can't know the version of the helm.

Reply:
1. I know that most linux distributions will install curl by default, macos may have issues, but the helm install script also relies on curl or wget commands. If there is no curl command, it will report an error and exit.here first check whether the curl command exists. If it does not exist, how about letting the user install it manually?
2. After installation, the heml version will be detected. Is there any problem here?

Reply:
For 1, I thought I've seen many folks using helmfile on busybox and alpine, which doesn't install curl by default. Also, it obviously doesn't work on Windows by default. It would still work on Linux subsystem for Windows which is based on Ubuntu, but still, I'd prefer using Go's own HTTP client for downloading the script. That way, the only external dependency becomes a (possibly posix-compliant) shell to run the installation script.

Reply:
For 2, yes, I think it's problematic because each Helmfile version supports only two Helm versions. For example, if you let an outdated Helmfile installation to use the latest Helm installed via the get-helm-3 script, there's no guarantee for it to work at all, as that Helmfile version wasn't tested with the latest Helm version.

Fortunately, the get-helm-3 script seems to accept DESIRED_VERSION for installing a specific version of Helm. Probably we can maintain a Golang `[]string` or any other data structure for holding the list of supported helm minor versions and let itself download the latest supported Helm version obtained from the list at the init time?

Reply:
> For 2, yes, I think it's problematic because, in reality, each Helmfile version supports only two Helm versions. For example, if you let an outdated Helmfile installation use the latest Helm installed via the get-helm-3 script, there's no guarantee for it to work at all, as that Helmfile version wasn't tested with the latest Helm version.
> 
> Fortunately, the get-helm-3 script seems to accept DESIRED_VERSION for installing a specific version of Helm. Probably we can maintain a Golang `[]string` or any other data structure for holding the list of supported helm minor versions and let itself download the latest supported Helm version obtained from the list at the init time?

We can define a recommended version of helm, just install the recommended version, what do you think?

Reply:
@xiaomudk Thanks for your understanding and efforts!

> We can define a recommended version of helm, just install the recommended version, what do you think?

Sounds great!
FYI, we currently maintain a set of supported dependencies' versions in a test matrix declared within ci.yaml. For example, see https://github.com/helmfile/helmfile/pull/434/filesex
I guess, we might just copy the version numbers declared in the matrix to a new golang package inside the helmfile pkg, perhaps somewhere like `pkg/externaldeps` a‘nd expose the latest recommended helm version supported by helmfile via a dedicated Go function like `func.GetLatestSupportedHelmVersion() semver.Version‘ within `pkg/externaldeps` so that you can call it from within the helmfile-init code.
WDYT?

Reply:
@mumoshu  What is the specific working logic of GetLatestSupportedHelmVersion? I'm declaring the version number in a constant now, which I think can achieve the same purpose.

Reply:
> I'm declaring the version number in a constant now

That would be 100% fine for the first implementation!
We may reconsider how the version numbers embedded in the binary/Go source be synced with those covered in the test matrix cc/ @yxxhero

Reply:
@mumoshu yes. I will try to do that.